applier/applier.go

```diff
@@ -8,32 +8,60 @@ import (
 )
 
 type Applier struct {
-	cache cache.LocalArtifactCache
+	cache []cache.LocalArtifactCache
 }
 
 func NewApplier(c cache.LocalArtifactCache) Applier {
+	return Applier{cache: []cache.LocalArtifactCache{c}}
+}
+
+func NewApplierMultiCache(c []cache.LocalArtifactCache) Applier {
 	return Applier{cache: c}
 }
 
+func (a Applier) mergeCacheLayers(layers map[types.CacheType]types.BlobInfo) types.BlobInfo {
+	base := layers[types.BuiltInCache]
+	for cache, layer := range layers {
+		if cache == types.BuiltInCache {
+			continue
+		}
+		if base.CustomResources == nil {
+			base.CustomResources = map[string]types.AnalyzerResources{}
+		}
+		base.CustomResources[string(cache)] = layer.CustomResources[string(cache)]
+	}
+	return base
+}
+
 func (a Applier) ApplyLayers(imageID string, diffIDs []string) (types.ArtifactDetail, error) {
 	var layers []types.BlobInfo
 	for _, diffID := range diffIDs {
-		layer, _ := a.cache.GetBlob(diffID)
-		if layer.SchemaVersion == 0 {
-			return types.ArtifactDetail{}, xerrors.Errorf("layer cache missing: %s", diffID)
+		// combining a1,b1,c1 = abc1
+		blobsInCache := map[types.CacheType]types.BlobInfo{}
+		for i := range a.cache {
```

Comment:
Do not iterate cache here since ApplyLayers is supposed to be called for each cache respectively.